$OpenBSD: patch-libproxy_CMakeLists_txt,v 1.6 2017/04/18 09:13:20 sthen Exp $

Adding -pthread:
https://code.google.com/p/libproxy/issues/detail?id=196&thanks=196&ts=1370457244

Setting _POSIX_C_SOURCE unsets __BSD_VISIBLE by default, and several symbols
end up being hidden, resulting in build failure with clang.
https://svnweb.freebsd.org/ports/head/net/libproxy/files/patch-libproxy_CMakeLists.txt?revision=406680&view=markup 

--- libproxy/CMakeLists.txt.orig	Mon Jan  9 14:22:22 2017
+++ libproxy/CMakeLists.txt	Sun Apr 16 22:16:52 2017
@@ -6,8 +6,7 @@ set_project_version(0 4 14)
 if (WIN32)
   add_definitions(-D_CRT_SECURE_NO_WARNINGS=1)
 else(WIN32)
-  add_definitions(-D_POSIX_C_SOURCE=200112L)
-  set(CMAKE_CXX_FLAGS "-fvisibility=hidden ${CMAKE_CXX_FLAGS}")
+  set(CMAKE_CXX_FLAGS "-fvisibility=hidden -pthread ${CMAKE_CXX_FLAGS}")
 endif(WIN32)
 
 ### Modules
